File: pilhas-filas/python/queue1.py
class Queue1:
    def __init__(self):
        self.list = []

    def is_empty(self):
        return len(self.list) == 0

    def add(self, item):
        self.list.append(item)

    def remove(self):
        if self.is_empty():
            raise Exception("Queue is empty")
        return self.list.pop(0)

    def peek(self):
        if self.is_empty():
            raise Exception("Queue is empty")
        return self.list[0]

    def count(self):
        return len(self.list)

File: pilhas-filas/java/src/Queue1.java
import java.util.ArrayList;
import java.util.List;

public class Queue1<T> {

	private List<T> list;

    public Queue1() {
        this.list = new ArrayList<>();
    }

    public boolean isEmpty() {
        return this.list.isEmpty();
    }

    public void add(T item) {
        this.list.add(item);
    }

    public T remove() {
        if (this.isEmpty()) {
            throw new IllegalStateException("Queue is empty");
        }
        return this.list.remove(0);
    }

    public T peek() {
        if (this.isEmpty()) {
            throw new IllegalStateException("Queue is empty");
        }
        return this.list.get(0);
    }

    public int count() {
        return this.list.size();
    }
}
